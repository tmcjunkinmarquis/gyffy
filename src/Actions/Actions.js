export const inputLocation = (location)=>({
  type: 'INPUT_LOCATION',
  location
});

export const loadStations = (stations)=>({
  type: 'LOAD_STATIONS',
  stations
});