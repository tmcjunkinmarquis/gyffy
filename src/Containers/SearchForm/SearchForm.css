.form {
  margin-left: 10px;
  margin-top: 25px;
}

input {
  height: 40px;
  width: 250px;
}

button {
  margin-left: 20px;
  box-shadow: 1px 1px 3px #6A9ECF;
  height: 45px;
  width: 45px;
}