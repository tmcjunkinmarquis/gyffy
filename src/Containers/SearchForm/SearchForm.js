import React, { Component } from 'react';
import { connect } from 'react-redux';
import './SearchForm.css';
import { inputLocation, loadStations } from '../../Actions/Actions';
import PropTypes from 'prop-types';
import { locationFetch } from '../../apiCalls/apiCalls';
import { cleanerForStationsByZip } 
  from '../../SampleDataAndCleaner/cleanerForStationsByZip';
export class SearchForm extends Component{
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    };
  }

  handleChange = (event)=>{
    this.setState({
      value: event.target.value
    }); 
  }

  resetForm = () => {
    this.setState({ value: '' });
  }

  makeStationFetch = async ()=>{
    const unrefinedStationData = await locationFetch(this.state.value);
    const refinedStationData = cleanerForStationsByZip(unrefinedStationData);
    this.props.loadStoreWithStations(refinedStationData); 
  }

  handleSubmitForm = (event) => {
    event.preventDefault();
    this.props.handleSubmit(this.state.value);
    this.resetForm();
    this.makeStationFetch(); 
  }

  render() {
    return (
      <form 
        action="" 
        onSubmit={this.handleSubmitForm}
        className= "search-form"
      >
        <input 
          type="text"
          value={this.state.value}
          placeholder='Type zipcode'
          onChange = {this.handleChange}
        />
        <button>Submit</button>
      </form>
    );
  } 
}

export const mapStateToProps = (state)=>({
  location: state.location
});

export const mapDispatchToProps = (dispatch) => ({
  handleSubmit: (location) => dispatch(inputLocation(location)),
  loadStoreWithStations: (stations)=> dispatch(loadStations(stations))
});

export default connect(
  mapStateToProps, mapDispatchToProps
)(SearchForm);

SearchForm.propTypes ={
  handleSubmit: PropTypes.func,
  location: PropTypes.string,
  loadStoreWithStations: PropTypes.func
};
