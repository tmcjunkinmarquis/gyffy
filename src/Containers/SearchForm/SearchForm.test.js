
import React from 'react';
import { SearchForm, mapStateToProps, mapDispatchToProps } from './SearchForm';
import * as action from '../../Actions/Actions';
import { shallow } from 'enzyme';
// jest.mock('../../apiCalls/apiCalls');


describe('', ()=>{
  let wrapper;
  let mockHandleSubmitForm;
  let mockEvent;
  let mockHandleSubmit;
  let mockLoadStoreWithStations;
  let mockProps;

  beforeEach(()=>{
    mockEvent = { preventDefault: jest.fn(), target: { value: 'Denver' }};
    mockHandleSubmit = jest.fn();
    mockHandleSubmitForm = jest.fn();
    mockLoadStoreWithStations = jest.fn();
    mockProps = {
      action: '',
      className: 'search-form',
      location: '61615'
    }
    wrapper = shallow(
      <SearchForm
        onSubmit = { mockHandleSubmitForm }
        handleSubmit = {mockHandleSubmit}
        loadStoreWithStations={mockLoadStoreWithStations}
        {...mockProps}
      />);
  });
  
  
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it.skip('should call handleSubmit upon calling handleSubmitForm', () => {
    const mockInitialState = { value: ''};
    wrapper.setState({mockInitialState});

    expect(wrapper.state('value')).toEqual('');

    wrapper.find('.search-form').simulate('submit', mockEvent);
    
    expect(wrapper.instance().props.handleSubmit).toHaveBeenCalled();
    // expect(wrapper.state().value.target.value).toEqual('Denver')
  });

  

  describe('mapDispatchToProps', ()=>{
    let mockDispatch;
    
    beforeEach(()=>{
      mockDispatch = jest.fn();
    });

    it('should call dispatch when using handleSubmit from mdtp', () => {
      const actionToDispatch = action.inputLocation('Denver');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.handleSubmit('Denver');

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);  
    });

    it('should call dispatch when using loadStoreWithStations from mdtp', () => {
      const actionToDispatch = action.loadStations([{name: 'bubbas'}, {name: 'bettys'}]);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.loadStoreWithStations([{ name: 'bubbas' }, { name: 'bettys' }]);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
