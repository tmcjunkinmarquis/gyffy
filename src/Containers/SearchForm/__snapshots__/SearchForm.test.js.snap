// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchForm
    handleSubmit={[Function]}
    onSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "action": "",
      "children": Array [
        <input
          onChange={[Function]}
          placeholder="Type location"
          type="text"
          value=""
/>,
        <button>
          Submit
</button>,
      ],
      "className": "search-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Type location",
          "type": "text",
          "value": "",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Submit",
        },
        "ref": null,
        "rendered": "Submit",
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "action": "",
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Type location"
            type="text"
            value=""
/>,
          <button>
            Submit
</button>,
        ],
        "className": "search-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Type location",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
